### PR TITLE
initial commit, adds basic repo structure

### PR DESCRIPTION
**Pull request by aristidesstaffieri**
[Ticket](https://stellarorg.atlassian.net/jira/software/c/projects/WAL/boards/36?modal=detail)
This adds the basic repo layout/patterns.

The setup is a monorepo in the spirit of [scaffold-eth](https://github.com/scaffold-eth/scaffold-eth/tree/master).
The basic idea is that we can organize each example under the `./modules` directory, and each module that can be a standalone dapp can be a [git submodule](https://git-scm.com/book/en/v2/Git-Tools-Submodules) to another repo which allows for better "fork-ability" as well as a yarn workspace under the root. Each module that does not need to be forkable or a standalone example can just be a yarn workspace.

Right now we have the following structure, with placeholders for the two planned modules.
```
.
├── README.md
├── modules
│   ├── mint-token
│   │   ├── README.md
│   │   └── package.json
│   ├── payment
│   │   ├── README.md
│   │   └── package.json
├── package.json
└── yarn.lock
```

Open Questions -

Should we commit to Typescript throughout? If so, we can add a top level build step for convenience.

### CODE REVIEW

**Comment by Julian-dev28** [comment]
@aristidesstaffieri I think we should commit to typescript throughout

**Comment by piyalbasu** [comment]
Agreed on Typescript. If you're prototyping in JS, it's easy to strip out types if needed. If you're prototyping in TS, it can be kinda annoying if you're trying to copy/paste JS and you start getting missing type errors

File: modules/shared/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# shared
```
Comment (piyalbasu) [inline modules/shared/README.md:1]: I think we should be careful about using a `shared` folder too much. Sometimes a folder like this in a monorepo makes code more DRY, but it makes it a little harder to find things at first glance (Freighter repo, for ex)

imo, we should err on the side of being repetitive and verbose in the code just to make things very clear

**Review by piyalbasu — APPROVED** [review]

File: modules/shared/README.md (hunk 1 above)
Comment (aristidesstaffieri) [reply modules/shared/README.md:1]: Yeah maybe it also makes forking projects a bit weird if your helpers are all in some other repo's submodule. We could drop the shared folder/workspace in favor of more repetition and hopefully some of the repetitive things can make their way back to the wallet libs.

**Pull request merged**